src/utils/components/PodSelectorModal/SelectorInput.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { useEffect, useRef, useState } from 'react';
+import TagsInput from 'react-tagsinput';
+import classNames from 'classnames';
+
+import { Label as PfLabel } from '@patternfly/react-core';
+
+import { cleanSelectorStr, cleanTags, isTagValid } from './selectorUtils';
+
+interface SelectorInputProps {
+  autoFocus?: boolean;
+  inputProps?: React.InputHTMLAttributes<HTMLInputElement>;
+  labelClassName?: string;
+  onChange: (tags: string[]) => void;
+  options?: { basic?: boolean };
+  placeholder?: string;
+  tags: string[];
+}
+
+const SelectorInput: React.FC<SelectorInputProps> = ({
+  autoFocus,
+  inputProps,
+  labelClassName,
+  onChange,
+  options = {},
+  placeholder,
+  tags: initialTags,
+}) => {
+  const isBasic = !!options.basic;
+  const [inputValue, setInputValue] = useState('');
+  const [tags, setTags] = useState<string[]>(initialTags);
+  const ref = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (!initialTags.every((tag, index) => tag === tags[index])) {
+      setTags(initialTags);
+    }
+  }, [initialTags, tags]);
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const value = e.target.value;
+    setInputValue(value);
+  };
+
+  const handleChange = (newTagsInput: string[], changed: string[]) => {
+    const newTag = changed[0];
+
+    if (!isTagValid(newTag, isBasic)) {
+      return;
+    }
+
+    const cleanNewTag = cleanSelectorStr(newTag);
+
+    if (newTagsInput.filter((tag) => tag === cleanNewTag).length > 1) {
+      return;
+    }
+
+    const newTags = cleanTags(newTagsInput);
+    setInputValue('');
+    setTags(newTags);
+    onChange(newTags);
+  };
+
+  const inputPropsWithDefaults = {
```

Comment:
can be in util file

Reply:
what do you mean?

Reply:
just 3 props are static. The other once depends on the component state